### ISSUE TITLE
Support for entryComponents?

### ISSUE THREAD

**Issue by thomasbottonari** [ui-jar/ui-jar#35]
I am trying to add a component to my UI-Jar project that creates another component dynamically, and as such requires that I add that dynamically created component to the module's `entryComponents` array. Obviously there isn't support for `entryComponents` in `TestBed.configureTestingModule`, but I've had success in the past in my normal unit test spec files by doing the following after calling `compileComponents()`.

```
TestBed.overrideModule(BrowserDynamicTestingModule, {
    set: {
        entryComponents: [ SomeDynamicComponent ]
    }
});
```

I tried doing that in my ui-jar specfile but am still getting the following error:
```
Error: No component factory found for SomeDynamicComponent. Did you add it to @NgModule.entryComponents?
```

Is this intended to be supported? Any insight into why it is not working, or what I could do differently?

**Comment by nording** [comment]
Thanks for the detailed issue! For now that is not supported yet, but it should be added. Sounds like a great thing to support!

**Comment by Karankang007** [comment]
I was able to get this working. Basically, you have to create another TestEntryModule with the declaration,export, and entry components.

And then import this TestEntryModule in your TestModule.

**Comment by nording** [comment]
@Karankang007 nice tips! I did not think about that solution :)

**Comment by thomasbottonari** [comment]
Thanks for the quick response. I had tried this myself but kept getting an error stating `Uncaught ReferenceError: PopoverModule is not defined`, even though I am clearly defining it in the same file. Below is the snippet of code in question. Do you see any issues?

```
@NgModule({
    declarations: [
        PopoverContentComponent,
        PopoverDirective
    ],
    exports: [
        PopoverContentComponent,
        PopoverDirective
    ],
    entryComponents: [ PopoverContentComponent ]
})
export class PopoverModule { }

describe('IconPopoverComponent Example', () => {
    let fixture: ComponentFixture<IconPopoverTestHost>;
    let component: IconPopoverTestHost;


    beforeEach(async(() => {
        /**
         * @uijar IconPopoverComponent
         * @hostcomponent IconPopoverTestHost
         */
        TestBed.configureTestingModule({
            imports: [
                CommonModule,
                AngularSvgIconModule,
                HttpClientModule,
                PopoverModule
            ],
            declarations: [
                IconComponent,
                IconPopoverTestHost,
                IconPopoverComponent,
            ]
        }).compileComponents().then(() => {
            fixture = TestBed.createComponent(IconPopoverTestHost);
            component = fixture.componentInstance;
        });
    }));
```

**Comment by nording** [comment]
@thomasbottonari That's a bug, I created #36. 
A workaround is to declare your "PopoverModule" outside the test file, then it will be defined correct.

**Comment by thomasbottonari** [comment]
Great, this worked for me. Thank you!

**Comment by nording** [comment]
Support for "entryComponents" is going to be available in next release!

**Comment by nording** [comment]
A new release is available at npm (v1.1.5) with a fix for this issue.
